pkg/expression/constant_propagation.go

```diff
@@ -356,15 +358,19 @@ func (s *propConstSolver) propagateColumnEQ() {
 				cond := s.conditions[k]
 				replaced, _, newExpr := tryToReplaceCond(s.ctx, coli, colj, cond, false)
 				if replaced {
-					if s.pushDownfilter != nil && !s.pushDownfilter(newExpr) {
-						continue
+					if s.vaildExprFunc != nil {
+						if !s.vaildExprFunc(newExpr) {
+							continue
+						}
 					}
 					s.conditions = append(s.conditions, newExpr)
 				}
 				replaced, _, newExpr = tryToReplaceCond(s.ctx, colj, coli, cond, false)
 				if replaced {
-					if s.pushDownfilter != nil && !s.pushDownfilter(newExpr) {
-						continue
+					if s.vaildExprFunc != nil {
+						if !s.vaildExprFunc(newExpr) {
```

Comment:
why nested if instead of the original one if branch?

Reply:
It might just look visually unappealing; there's actually no particular reason.

Reply:
restore it.